salt/modules/debian_ip.py

```diff
@@ -899,17 +899,18 @@ def _parse_settings_bond_1(opts, iface, bond_def):
             _log_default_iface(iface, binding, bond_def[binding])
             bond.update({binding: bond_def[binding]})
 
-    if 'use_carrier' in opts:
-        if opts['use_carrier'] in _CONFIG_TRUE:
-            bond.update({'use_carrier': '1'})
-        elif opts['use_carrier'] in _CONFIG_FALSE:
-            bond.update({'use_carrier': '0'})
+    if __grains__['os'] == "Ubuntu" and __grains__['osrelease_info'][0] >= 16:
```

Comment:
Did you mean for this to be `<= 16` instead of `>=`? As this stands, this will cause the `if 'use_carrier' in opts` block to be skipped for older versions of Ubuntu, which is a change in behavior that I don't think you're intending based on your description of the issue.

I am also curious if this block is used for Debian boxes and if only checking for `Ubuntu` releases, this will cause a regression.